crates/lang_srv/src/analysis/completion.rs

```diff
@@ -0,0 +1,441 @@
+    interns: &Interns,
+    completions: Vec<(String, Variable)>,
+) -> Vec<CompletionItem> {
+    completions
+        .into_iter()
+        .map(|(symbol, var)| make_completion_item(subs, module_id, interns, symbol, var))
+        .collect()
+}
+
+///Gets completion items for a record field
+///Uses
+fn find_record_fields(var: Variable, subs: &mut Subs) -> Vec<(String, Variable)> {
+    let content = subs.get(var);
+    match content.content {
+        roc_types::subs::Content::Structure(typ) => match typ {
+            roc_types::subs::FlatType::Record(fields, ext) => {
+                let field_types = fields.unsorted_iterator(subs, ext);
+                let fields: Vec<_> = match field_types {
+                    Ok(field) => field
+                        .map(|a| {
+                            let var = match a.1 {
+                                roc_types::types::RecordField::Demanded(var)
+                                | roc_types::types::RecordField::Required(var)
+                                | roc_types::types::RecordField::Optional(var)
+                                | roc_types::types::RecordField::RigidRequired(var)
+                                | roc_types::types::RecordField::RigidOptional(var) => var,
+                            };
+                            (a.0.clone().into(), var)
+                        })
+                        .collect(),
+                    Err(err) => {
+                        eprintln!(
+                            "WARN:Error getting record field types for completion{:?}",
+                            err
+                        );
+                        vec![]
+                    }
+                };
+                fields
+            }
+            roc_types::subs::FlatType::Tuple(elems, ext) => {
+                let elems = elems.unsorted_iterator(subs, ext);
+                let elems: Vec<_> = match elems {
+                    Ok(elem) => elem.map(|(num, var)| (num.to_string(), var)).collect(),
+                    Err(err) => {
+                        eprintln!("WARN:Error getting tuple elems for completion{:?}", err);
+                        vec![]
+                    }
+                };
+                elems
+            }
+
+            _ => {
+                eprintln!(
+                    "WARN: Trying to get field completion for a type that is not a record   ",
+                );
+                vec![]
+            }
+        },
+        roc_types::subs::Content::Error => {
+            //This is caused by typechecking our partially typed variable name causing the typechecking to be confused as the type of the parent variable
+            //TODO! ideally i could recover using some previous typecheck result that isn't broken
+            eprintln!("ERROR: variable type of record was of type error cannot access field",);
+            vec![]
+        }
+        a => {
+            eprintln!("variable before field was unsuported type:{:?}", a);
```

Comment:
Wouldn't that make it only run in debug? The whole reason i put this log in is because it shouldn't happen and if it does hopefully someone will tell us. If you disagree I'm happy to change that though.